python/swe/README.md

```diff
@@ -1,50 +1,126 @@
-# README for swe.py
+# SWE Development Kit
+
+## Table of Contents
+
+- [SWE Development Kit](#swe-development-kit)
+  - [Table of Contents](#table-of-contents)
+  - [Overview](#overview)
+  - [Dependencies](#dependencies)
+  - [Getting Started](#getting-started)
+    - [Scaffolding a New Agent](#scaffolding-a-new-agent)
+    - [Docker Environment](#docker-environment)
+    - [Running the Benchmark](#running-the-benchmark)
+  - [Extending Functionality](#extending-functionality)
+    - [Adding a New Tool](#adding-a-new-tool)
+    - [Adding a New Shell Tool](#adding-a-new-shell-tool)
 
 ## Overview
 
-The `swe.py` script is part of the Composio software engineering (SWE) agent framework.
-It is designed to automate tasks related to software development, including issue resolution, code reviews, and patch submissions using AI-driven agents.
+Composio-swe is a powerful framework for building and evaluating software engineering agents. With this toolkit, you can:
+
+- Build your own custom agents
+- Utilize pre-built agents provided by Composio
+- Run the SWE-Bench benchmark to evaluate agent performance
 
 ## Dependencies
 
-1. Docker Desktop should be installed.
-2. Get the Github Access Token.
-3. Install the dependencies using `pip install -r requirements.txt`.
-4. Add the LLM configuration via `OPENAI_API_KEY` or `ANTHROPIC_API_KEY` or (`AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT`) environment variables.
-5. Add the environment variable `export GITHUB_ACCESS_TOKEN = <git_access_token>`.
-6. If you want to use Helicone, add the environment variable `export HELICONE_API_KEY = <helicone_api_key>`.
+Before getting started, ensure you have the following set up:
+
+1. **Installation**:
+
+   ```
+   pip install composio-swe composio-core
+   ```
+
+2. **Optional CrewAI Support**:
+   If you plan to use CrewAI for your agent:
+
+   ```
+   pip install crewai composio-crewai
+   ```
+
+3. **GitHub Access Token**:
+
+   - Create one at https://github.com/settings/tokens with necessary permissions
+   - Set the environment variable: `export GITHUB_ACCESS_TOKEN=<your_token>`
+
+4. **LLM Configuration**:
+   - Set up OpenAI API key: `export OPENAI_API_KEY=<your_key>`
+   - Other LLMs can be configured by modifying the code (details provided later)
+
+## Getting Started
+
+### Scaffolding a New Agent
+
+1. Run the following command:
+
+   ```
+   composio-swe scaffold crewai -o <path>
+   ```
+
+   This creates a new agent in `<path>/agent` with four key files:
+
+   - `main.py`: Entry point to run the agent on your issue
+   - `agent.py`: Core agentic logic (edit this to customize behavior)
+   - `prompts.py`: Agent prompts
```

Comment:
```suggestion
   - `prompts.py`: Default agent prompts
```

Reply:
Agent prompts, if the user changes they can become non-default.